html_app/components/MicroscopyModelFactory.js

```diff
@@ -23,75 +23,73 @@ scb.components.MicroscopyModelFactory = function scb_components_MicroscopyModelF
             var imgs = [];
             var isFound = false;
             var max;
-            var index, slide_array, alreadySelected, number_of_comparisons;
+            var index, alreadySelected, number_of_comparisons;
+            var slide_array=[];
             var parser = model.slide;
 
-            if (parser.parser_simple){
+            if (parser.parser_simple) {
 
-
-                    var micro_state = {
-                        kind: function (str) {
-                            return str == slide_type;
-                        },
-                        collection_id: function (str) {
-                            return str == collection_id;
-                        },
-                        drug_id: function (str) {
-                            return str == drug_id;
-                        },
-                        cell_line: function (str) {
-                            return str == cell_line;
-                        },
-                        conditions: function (str) {
-                            return str == conditions;
+                var micro_state = {
+                    kind: function (str) {
+                        return str == slide_type;
+                    },
+                    collection_id: function (str) {
+                        return str == collection_id;
+                    },
+                    drug_id: function (str) {
+                        return str == drug_id;
+                    },
+                    cell_line: function (str) {
+                        return str == cell_line;
+                    },
+                    conditions: function (str) {
+                        return str == conditions;
+                    }
+                };
+                _.each(parser.parser_simple, function (rule) {
+                    var matches = true;
+                    _.each(rule.match, function (property) {
+                        if (micro_state[property]) {
+                            matches &= micro_state[property](rule[property]);
+                        } else {
+                            console.info("UNDEFINED PROPERTY: " + property);
                         }
-                    };
-                    _.each(parser.parser_simple, function (rule) {
-                            var matches = true;
-                            _.each(rule.match, function (property) {
-                                if (micro_state[property]) {
-                            	    matches &= micro_state[property](rule[property]);
-                                }else {
-                            	    console.info("UNDEFINED PROPERTY: " + property);
+                    });
+                    if (!isFound) {
+                        if (matches) {
+                            if (template.slide_parser) {
+                                max = template.slide_parser[collection_id][slide_type][conditions].length;
```

Comment:
There are a couple of these `template.slide_parser[collection_id][slide_type][conditions]`. I think it is worth assigning that to a variable and using it instead of fully indexing it on each use for this and line 70, but feel free to ignore if you have a preference.